keps/sig-testing/4224-replace-kubeup-clusters-with-kops/README.md

```diff
@@ -0,0 +1,429 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Kubernetes 1.0 was released in July 2015 and since then, we have been using a collection of shell scripts in the cluster/ folder of the kubernetes/kubernetes repository to create Kubernetes clusters for e2e testing. These are called kube-up clusters and are deployed with fragile shell scripts that have caused many problems over the years. They have been deprecated for a long time and it is time to replace them with a different cluster provisioning tool.
+
+Most of the Kubernetes e2e tests have to run on a real cluster on a cloud provider. The Kubernetes project has several 1st party tools that provision clusters on GCP such as kops, CAPI and kubespray. We could also reimplement kube-up in go.
+
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+- The shell scripts are very fragile and no new features are being accepted
+- arm64 testing needs to be done on GCE
+- python and debian upgrades always break the Kubernetes e2e tests, particularly at a bad time(cutting releases, reverting/patching a critical bug). This will no longer be the case once we move to kops. For example, 
+- We have some tests in kubernetes that make assumptions about a specific pieces of cloud infrastructure and are not a good fit for kubernetes e2e tests. Tests that rely on cloud-infra that are not reachable will be removed.
```

Comment:
> Tests that rely on cloud-infra that are not reachable will be removed.

can you please expand on this one?

Reply:
I think he means things like these https://github.com/kubernetes/kubernetes/pull/120968